File: easy/Complement of Base 10 Integer/Go/main.go
// Problem: Complement of Base 10 Integer
// Link to the problem: https://leetcode.com/problems/complement-of-base-10-integer/
func bitwiseComplement(n int) int {
	if n == 0 {
		return 1
	}
	var x int = 1
	for x < n {
		x = (x << 1) | 1
	}
	var ans int = n ^ x
	return ans
}

File: easy/Complement of Base 10 Integer/Rust/main.rs
// Problem: Complement of Base 10 Integer
// Link to the problem: https://leetcode.com/problems/complement-of-base-10-integer/
impl Solution {
    pub fn bitwise_complement(n: i32) -> i32 {
        if (n == 0) {
            return 1;
        }
        let mut x: i32 = 1;
        while (x < n) {
            x = (x << 1) | 1;
        }
        let ans: i32 = n ^ x;
        return ans;
    }
}
